commands/certs/add.js

```diff
@@ -113,25 +113,69 @@ function * getChoices (certDomains, newDomains, existingDomains, context) {
   }
 }
 
+function * getDomains (context, heroku) {
+  function someNull (domains) {
+    return _.some(domains, (domain) => domain.kind === 'custom' && domain.cname == null)
+  }
+
+  function * apiRequest (context, heroku) {
+    return yield heroku.request({
+      path: `/apps/${context.app}/domains`
+    })
+  }
+
+  let apiDomains = yield apiRequest(context, heroku)
+
+  if (someNull(apiDomains)) {
+    yield cli.action('Waiting for stable domains to be created', co(function * () {
+      const wait = require('co-wait')
+
+      let i = 0
+      do {
+        // trying 30 times was easier for me to test that setTimeout
+        if (i >= 30) {
+          throw new Error('Timed out while waiting for stable domains to be created')
+        }
+
+        yield wait(1000)
+        apiDomains = yield apiRequest(context, heroku)
```

Comment:
or if you really do want to put that in a function, it could just be

```js
function apiRequest (context, heroku) {
  return heroku.get(`/apps/${context.app}/domains`)
}
```

Also, in the co docs it says it isn't a good idea to yield directly on a generator without wrapping it in `co.wrap`.